### ISSUE TITLE
Pylance semantic highlighting not working on dev container

### ISSUE THREAD

**Issue by nickolasclarke** [microsoft/pylance-release#2803]
Issue Type: <b>Bug</b>

When I load the same file in both a Dev Container or on my host, pylance semantic highlighting only appears to work on the host instance of VSCode. All settings seem to be the same, with the python and pylance extensions installed both locally and on the dev container. 

![image](https://user-images.githubusercontent.com/2177320/168006288-a8bcb3cc-09a7-4c2e-b95d-a7a271f412d6.png)

Extension version: 2022.5.1
VS Code version: Code 1.67.1 (da15b6fd3ef856477bf6f4fb29ba1b7af717770d, 2022-05-06T12:37:14.619Z)
OS version: Darwin arm64 21.4.0
Restricted Mode: No
Remote OS version: Linux arm64 5.10.104-linuxkit

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 Pro (8 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|8, 7, 7|
|Memory (System)|16.00GB (0.07GB free)|
|Process Argv|--crash-reporter-id ac0a67bf-7338-4597-9ba8-417b7627d890|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|Dev Container: WG Workspace|
|OS|Linux arm64 5.10.104-linuxkit|
|CPUs|unknown (4 x 0)|
|Memory (System)|7.76GB (2.71GB free)|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256860
vstes627:30244334
pythonvspyl392:30443607
pythontb:30283811
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30445986
pythondataviewer:30285071
vscod805cf:30301675
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
vsc1dst:30438360
pythonvs932:30410667
wslgetstarted:30449410
pythonvsnew555:30457759
vscscmwlcmt:30465135
cppdebugcf:30475217
vscaat:30438848
vsclangdf:30486550

```

</details>

**Comment by bschnurr** [comment]
can you please upload the logs, see these docs
https://github.com/microsoft/pylance-release/blob/main/TROUBLESHOOTING.md#filing-an-issue

**Comment by nickolasclarke** [comment]
After setting `"python.analysis.logLevel": "Trace"` and reloading the window, semantic tracing does appear to be working now. If I can get this to reproduce I'll upload logs.

**Comment by CarlosDomingues** [comment]
I am having a similar issue.

I set `"python.analysis.logLevel": "Trace"`, but no logs are being shown[1]. 

When starting the development container, I also get a message saying claiming Pylance is not installed [2]. 

When trying to install it, looks like it was installed already[3]. 

I tried removing `"python.languageServer": "Pylance"` from my `.devcontainer/devcontainer.json` and falling back to `jedi` works as expected.

Is there any way to start Pylance manually?

**Comment by bschnurr** [comment]
no, you need the python extension's language server setting set to pylance
"python.languageServer": "Pylance"

double check pylance is installed
![image](https://user-images.githubusercontent.com/1946977/172209018-3bf82b7d-2c78-45a3-a3f9-cfaf863f10d7.png)

**Comment by plknkl** [comment]
for me the issue was specifying in `.devcontainer/devcontainer.json` the two extentions `"ms-python.python"` and `"ms-python.vscode-pylance"`. As soon as I removed the `"ms-python.vscode-pylance"` and left only `"ms-python.python"` everything started working as expected.

**Comment by mkaic** [comment]
For any other weary travelers who stumble across this thread: I was having this exact same issue (semantic highlighting working locally but not remotely) but none of the suggested fixes both in this thread and countless others I'd looked at did anything. After turning on trace-level logging as recommended here and looking at the outputs, I noticed Pylance was saying something about there being too many files to scan through in my project directory. 

I realized that the project directory contained a mountpoint for my multi-terabyte NAS containing millions of files, which would explain that error message. Basically, Pylance was trying to find the source code for the packages I was using so it could use that for enabling autocomplete and highlighting, but it couldn't because it was trying to recursively search through millions of files from the project's root directory. The error suggested I write a `pyrightconfig.json` file with an `exclude` field in it to tell Pylance it didn't have to scan the large number of nested directories within my project. I tried that but sadly wasn't able to get it to work just due to there being a *ton* of directories scattered all over the place in my project root. 

What ended up fixing the problem was just changing my VSCode workspace folder to one of the smaller subfolders (the one where I kept my code anyways), at which point highlighting and autocomplete were both back online. I hope this is helpful to someone else in the future.
